tools/rebasehelpers/commitchecker/validate.go

```diff
@@ -3,19 +3,53 @@ package main
 import (
 	"bytes"
 	"fmt"
-	"os/exec"
 	"regexp"
 	"strings"
 	"text/template"
 
 	"github.com/openshift/origin/tools/rebasehelpers/util"
 )
 
-var CommitSummaryErrorTemplate = `
-The following UPSTREAM commits have invalid summaries:
+var (
+	// AllCommitValidators holds all registered checks.
+	AllCommitValidators = []func(util.Commit) []string{
+		ValidateCommitAuthor,
+
+		// vendor/* -> commit meesgae
+		ValidatePatchHasUpstreamCommitMessage,
+		ValidateBumpHasBumpCommitMessage,
+		ValidateNoBumpAndPatchesTogether,
+
+		// commit message -> vendor/*
+		ValidateUpstreamCommit,
+		ValidateBumpCommit,
+	}
+)
+
+func ValidateCommitAuthor(commit util.Commit) []string {
+	var allErrors []string
+
+	if strings.HasPrefix(commit.Email, "root@") {
+		allErrors = append(allErrors, fmt.Sprintf("Commit %s has invalid email %q", commit.Sha, commit.Email))
```

Comment:
why strings for errors?

Reply:
errors may not start with capital letter